File: code2map/docs/examples/java/output/parts/UserManagementService_validateUserId.java
// code2map fragment (non-buildable)
// original: docs/examples/java/UserManagementService.java
// lines: 168-176
// symbol: UserManagementService#validateUserId
// notes: references java.util.ArrayList, java.util.HashMap, java.util.List, java.util.Map, java.util.Optional, java.util.regex.Pattern; calls isEmpty, userId.length, userId.trim
    private void validateUserId(String userId) {
        if (userId == null || userId.trim().isEmpty()) {
            throw new IllegalArgumentException("ユーザーIDは必須です");
        }
        if (userId.length() > MAX_USER_ID_LENGTH) {
            throw new IllegalArgumentException(
                "ユーザーIDは" + MAX_USER_ID_LENGTH + "文字以内で入力してください");
        }
    }

File: code2map/docs/examples/python/output/parts/UserManagementService__validate_user_id.py
# code2map fragment (non-buildable)
# id: CD14
# original: docs/examples/python/user_management_service.py
# lines: 205-209
# symbol: UserManagementService#_validate_user_id
# notes: references dataclasses.dataclass, re, typing.Optional; calls ValueError, len, user_id.strip
    def _validate_user_id(self, user_id: str) -> None:
        if not user_id or not user_id.strip():
            raise ValueError("ユーザーIDは必須です")
        if len(user_id) > self.MAX_USER_ID_LENGTH:
            raise ValueError(f"ユーザーIDは{self.MAX_USER_ID_LENGTH}文字以内で入力してください")
